File: src/dataset/cpp/triangle_exercism-stungeye/gemini/GeminiFlash_triangle_v4.cpp
#include "triangle.h"
#include <algorithm>
#include <cmath>

namespace triangle {
  flavor kind(double a, double b, double c) {
    if (!(a > 0 && b > 0 && c > 0)) {
      throw std::domain_error("All triangle sides must have a length greater than zero.");
    }

    double sides[3] = {a, b, c};
    std::sort(sides, sides + 3);

    if (!(sides[0] + sides[1] > sides[2])) {
      throw std::domain_error("Combined length of any two triangle sides much be greater or equal to length of third side.");
    }

    if (a == b) {
        if (b == c) {
            return flavor::equilateral;
        } else {
            return flavor::isosceles;
        }
    } else if (b == c || a == c) {
        return flavor::isosceles;
    }

    return flavor::scalene;
  }
}  // namespace triangle

File: src/dataset/javascript/triangle_exercism-elenamountz/gemini/GeminiFlash_triangle_v4.js
export default class Triangle{
  constructor(s1, s2, s3){
      this.s1 = s1;
      this.s2 = s2;
      this.s3 = s3;
  }
  kind(){
    const { s1, s2, s3 } = this;

    if (s1 <= 0 || s2 <= 0 || s3 <= 0) {
      throw new Error();
    }

    if (s1 + s2 <= s3 || s2 + s3 <= s1 || s3 + s1 <= s2) {
      throw new Error();
    }

    if (s1 === s2) {
      if (s1 === s3) {
        return 'equilateral';
      }
      return 'isosceles';
    } else if (s1 === s3 || s2 === s3) {
      return 'isosceles';
    } else {
      return 'scalene';
    }
  }
};